opensearch-operator/pkg/builders/cluster.go

```diff
@@ -531,19 +531,25 @@ func NewHeadlessServiceForNodePool(cr *opsterv1.OpenSearchCluster, nodePool *ops
 
 func NewServiceForCR(cr *opsterv1.OpenSearchCluster) *corev1.Service {
 
+	var configChecksum string
+
 	labels := map[string]string{
 		helpers.ClusterLabel: cr.Name,
 	}
+	annotations := map[string]string{
+		ConfigurationChecksumAnnotation: configChecksum,
+	}
```

Comment:
Why did you set the annotations to `configChecksum`, which is not even calculated?
The `GeneralConfig` already has an `Annotations` field, so I'd say just add those to the service spec.

Reply:
I can't see the annotation field `GeneralConfig` what's the field name in the struct?

Reply:
Ah, sorry, my bad, was looking at the NodePool struct. In this case: Add a new field `Annotations` to the `GeneralConfig` like there is at `NodePool`, then use the contents of that for the service annotations. While you're at it could you also extend `NewHeadlessServiceForNodePool` to add annotations to the service (a combination of the annotations from general and nodepool)?

Reply:
Alright I will do that

Reply:
@swoehrl-mw Done could you please review?